File: cdk-from-cfn-testing/expected/resource_w_json_type_properties/typescript/JsonPropsConstruct.ts
import * as cdk from 'aws-cdk-lib';
import * as iam from 'aws-cdk-lib/aws-iam';
import * as sqs from 'aws-cdk-lib/aws-sqs';
import { Construct } from 'constructs';

export interface JsonPropsConstructProps {
}

export class JsonPropsConstruct extends Construct {
  public constructor(scope: Construct, id: string, props: JsonPropsConstructProps = {}) {
    super(scope, id);

    // Resources
    const myQueue1 = new sqs.CfnQueue(this, 'MyQueue1', {
    });

    const myQueue2 = new sqs.CfnQueue(this, 'MyQueue2', {
    });

    const myRdMessageQueueGroup = new iam.CfnGroup(this, 'MyRDMessageQueueGroup', {
      policies: [
        {
          policyName: 'MyQueueGroupPolicy',
          policyDocument: {
            Statement: [
              {
                Effect: 'Allow',
                Action: [
                  'sqs:DeleteMessage',
                  'sqs:ReceiveMessage',
                ],
                Resource: [
                  myQueue1.attrArn,
                  myQueue2.attrArn,
                ],
              },
            ],
          },
        },
      ],
    });
  }
}

File: cdk-from-cfn-testing/expected/resource_w_json_type_properties/csharp/JsonPropsConstruct.cs
using Amazon.CDK;
using Amazon.CDK.AWS.IAM;
using Amazon.CDK.AWS.SQS;
using Constructs;
using System.Collections.Generic;

namespace JsonPropsConstruct
{
    public class JsonPropsConstructProps
    {
    }

    public class JsonPropsConstruct : Construct
    {
        public JsonPropsConstruct(Construct scope, string id, JsonPropsConstructProps props = null) : base(scope, id)
        {

            // Resources
            var myQueue1 = new CfnQueue(this, "MyQueue1", new CfnQueueProps
            {
            });
            var myQueue2 = new CfnQueue(this, "MyQueue2", new CfnQueueProps
            {
            });
            var myRdMessageQueueGroup = new CfnGroup(this, "MyRDMessageQueueGroup", new CfnGroupProps
            {
                Policies = new []
                {
                    new CfnGroup.PolicyProperty
                    {
                        PolicyName = "MyQueueGroupPolicy",
                        PolicyDocument = new Dictionary<string, object>
                        {
                            { "Statement", new []
                            {
                                new Dictionary<string, object>
                                {
                                    { "Effect", "Allow"},
                                    { "Action", new []
                                    {
                                        "sqs:DeleteMessage",
                                        "sqs:ReceiveMessage",
                                    }},
                                    { "Resource", new []
                                    {
                                        myQueue1.AttrArn,
                                        myQueue2.AttrArn,
                                    }},
                                },
                            }},
                        },
                    },
                },
            });
        }
    }
}
